### PR TITLE
Bug 1192275 - creating jsmarionette accessibility plugin.

### PR DESCRIPTION
**Pull request by yzen**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9744785fad03ba6389d4f3d68d24d2bc5668204

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f32f89eee881c87542791dbb493d4c9eeb21d7f3

File: .jshintignore [hunk 1]
```diff
@@ -74,6 +74,7 @@ tv_apps/tv-epg/bower_components/**
 tests/atoms/remote_date.js
 tests/atoms/screenshot.js
 tests/jsmarionette/client/marionette-client/docs/**
+tests/jsmarionette/plugins/marionette-a11y/lib/axe.min.js
```
Comment (gaye) [inline .jshintignore]: I can't seem to find this file in pr?

Comment (gaye) [inline .jshintignore]: Oh oops it wasn't shown since it's compiled/binary-looking

Comment (gaye) [inline .jshintignore]: Is there a reason this is minified?

File: tests/jsmarionette/plugins/marionette-a11y/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+   * Run an accessibility check of the current document.
+   * If there are violations, the check will assert.
+   */
+  check: function MarionetteA11y_check() {
+    this.client.executeAsyncScript(this.axe +
+      'axe.a11yCheck(document, {' +
+        'rules: {' +
+          '"document-title": { enabled: false },' +
+          '"meta-viewport": { enabled: false },' +
+          '"bypass": { enabled: false }' +
```
Comment (gaye) [inline tests/jsmarionette/plugins/marionette-a11y/index.js]: Why can't we append this to the `axe.js` file?

File: tests/jsmarionette/plugins/marionette-a11y/index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        '}' +
+      '}, function(results) {' +
+        'marionetteScriptFinished(results.violations);' +
+      '});',function(error, violations) {
+      if (error) {
```
Comment (gaye) [inline tests/jsmarionette/plugins/marionette-a11y/index.js]: There is also the sugary `assert.ifError(error);`

File: tests/jsmarionette/plugins/marionette-a11y/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        assert(false, 'Error running accessibility check');
+      }
+      if (violations.length > 0) {
+        assert(false, violations.map(function(violation) {
```
Comment (gaye) [inline tests/jsmarionette/plugins/marionette-a11y/index.js]: `assert(violatoins.length > 0)` ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=40208e650ef9abf9fc82c6378c057ef4b9fcf8b3

**Pull request closed without merging**